packages/flutter_tools/lib/src/macos/xcode.dart

```diff
@@ -185,3 +189,303 @@ class Xcode {
     );
   }
 }
+
+/// A utility class for interacting with Xcode xcdevice command line tools.
+class XCDevice {
+  XCDevice({
+    @required ProcessManager processManager,
+    @required Logger logger,
+    @required Xcode xcode,
+  }) : _processUtils = ProcessUtils(logger: logger, processManager: processManager),
+       _logger = logger,
+       _xcode = xcode;
+
+  final ProcessUtils _processUtils;
+  final Logger _logger;
+  final Xcode _xcode;
+
+  bool get isInstalled => _xcode.isInstalledAndMeetsVersionCheck && xcdevicePath != null;
+
+  String _xcdevicePath;
+  String get xcdevicePath {
+    if (_xcdevicePath == null) {
+      try {
+        _xcdevicePath = _processUtils.runSync(
+          <String>[
+            'xcrun',
+            '--find',
+            'xcdevice'
+          ],
+        ).stdout.trim();
+      } on ProcessException {
+        // Ignored, return null below.
```

Comment:
Consider printing a trace message with the exception object so we can get some additional info from a `--verbose` run.

Reply:
It looks like you may also want to pass `throwOnError: true` to `runSync()`. That will make `runSync()` throw a `ProcessException` if the `xcrun` exit code is non-zero.